### PR TITLE
Directory Component

### PR DESCRIPTION
**Pull request by g-luo**
* Create directory component
* Refactor show / hide functions for modals and versioning / directory component to be one function in script.js

### CODE REVIEW

File: project/src/main/java/com/google/sps/Servlets/FolderServlet.java [hunk 1]
```diff
@@ -25,13 +25,28 @@
 public class FolderServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (g-luo) [inline project/src/main/java/com/google/sps/Servlets/FolderServlet.java:27]: this function was approved in a previous pull request and I added it back exactly; feel free to just skim

**Review by g-luo — COMMENTED** [review]
Created a comment for reviewers.

**Review by tin0d0 — APPROVED** [review]
lgtm!

**Review by max-friedman — APPROVED** [review]
lgtm

**Review by HidyHan — APPROVED** [review]

**Review by frankpzn — APPROVED** [review]

**Pull request merged**